src/main/java/com/autotune/utils/MetricsConfig.java

```diff
@@ -0,0 +1,62 @@
+import io.micrometer.prometheus.PrometheusMeterRegistry;
+import io.micrometer.core.instrument.MeterRegistry;
+
+public class MetricsConfig {
+
+    private static MetricsConfig INSTANCE;
+    public static Timer timerlistRec, timerlistExp, timercreateExp, timerupdateResults;
+    public static Timer timerloadRecByExpName, timerloadResultsExpName, timerloadExpName;
+    public static Timer timerloadAllRec, timerloadAllExp, timerloadAllResults;
+    public static Timer timeraddRecDB, timeraddResultsDB, timeraddExpDB;
+
+    public static PrometheusMeterRegistry meterRegistry;
+
+    // Private constructor to prevent instantiation
+    private MetricsConfig() {
+        meterRegistry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
+        meterRegistry.config().commonTags("application", "Kruize");
+
+        timerlistRec = Timer.builder("kruizeAPI").description("Time taken for Kruize APIs").tag("api","listRecommendations").tag("method","GET").register(meterRegistry);
+        timerlistExp = Timer.builder("kruizeAPI").description("Time taken for Kruize APIs").tag("api","listExperiments").tag("method","GET").register(meterRegistry);
+        timercreateExp = Timer.builder("kruizeAPI").description("Time taken for Kruize APIs").tag("api","createExperiment").tag("method","POST").register(meterRegistry);
+        timerupdateResults = Timer.builder("kruizeAPI").description("Time taken for Kruize APIs").tag("api","updateResults").tag("method","POST").register(meterRegistry);
+
+        timerloadRecByExpName = Timer.builder("kruizeDB").description("Time taken for KruizeDB methods").tag("method","loadRecommendationsByExperimentName").register(meterRegistry);
+        timerloadResultsExpName = Timer.builder("kruizeDB").description("Time taken for KruizeDB methods").tag("method","loadResultsByExperimentName").register(meterRegistry);
+        timerloadExpName = Timer.builder("kruizeDB").description("Time taken for KruizeDB methods").tag("method","loadExperimentByName").register(meterRegistry);
+        timerloadAllRec = Timer.builder("kruizeDB").description("Time taken for KruizeDB methods").tag("method","loadAllRecommendations").register(meterRegistry);
+        timerloadAllExp = Timer.builder("kruizeDB").description("Time taken for KruizeDB methods").tag("method","loadAllExperiments").register(meterRegistry);
+        timerloadAllResults = Timer.builder("kruizeDB").description("Time taken for KruizeDB methods").tag("method","loadAllResults").register(meterRegistry);
+        timeraddRecDB = Timer.builder("kruizeDB").description("Time taken for KruizeDB methods").tag("method","addRecommendationToDB").register(meterRegistry);
+        timeraddResultsDB = Timer.builder("kruizeDB").description("Time taken for KruizeDB methods").tag("method","addResultsToDB").register(meterRegistry);
+        timeraddExpDB = Timer.builder("kruizeDB").description("Time taken for KruizeDB methods").tag("method","addExperimentToDB").register(meterRegistry);
```

Comment:
Please add appropriate descriptions for all of these

Reply:
I have kept the description the same for a metric (kruizeAPI and kruizeDB) purposefully.
The timers use the same metric name (kruizeAPI or kruizeDB) but have different tags based on the label it has used. If I use different descriptions for the same metric, it picks only the first description of the metric defined  and displays only that for all the tags of that metric which doesn't look appropriate.

For example, if I use 
```
timerloadRecByExpName = Timer.builder("kruizeDB").description("Time taken for loadRecommendationsByExperimentName method").tag("method","loadRecommendationsByExperimentName").register(meterRegistry);
timerloadResultsExpName = Timer.builder("kruizeDB").description("Time taken for loadResultsByExperimentName method").tag("method","loadResultsByExperimentName").register(meterRegistry);
```

Below is the /metrics data for that which picks the first variable description where the metric is defined. 

```
# HELP kruizeDB_seconds_max Time taken for loadRecommendationsByExperimentName method
# TYPE kruizeDB_seconds_max gauge
kruizeDB_seconds_max{application="Kruize",method="addRecommendationToDB",} 0.0
kruizeDB_seconds_max{application="Kruize",method="addResultsToDB",} 0.0
kruizeDB_seconds_max{application="Kruize",method="loadAllRecommendations",} 0.0
kruizeDB_seconds_max{application="Kruize",method="loadAllExperiments",} 0.0
kruizeDB_seconds_max{application="Kruize",method="addExperimentToDB",} 0.0
kruizeDB_seconds_max{application="Kruize",method="loadResultsByExperimentName",} 0.0
kruizeDB_seconds_max{application="Kruize",method="loadExperimentByName",} 0.0
kruizeDB_seconds_max{application="Kruize",method="loadAllResults",} 0.0
kruizeDB_seconds_max{application="Kruize",method="loadRecommendationsByExperimentName",} 0.0
# HELP kruizeDB_seconds Time taken for loadRecommendationsByExperimentName method
# TYPE kruizeDB_seconds summary
kruizeDB_seconds_count{application="Kruize",method="addRecommendationToDB",} 0.0
kruizeDB_seconds_sum{application="Kruize",method="addRecommendationToDB",} 0.0
kruizeDB_seconds_count{application="Kruize",method="addResultsToDB",} 0.0
kruizeDB_seconds_sum{application="Kruize",method="addResultsToDB",} 0.0
```